Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/HttpUtility.cs

```diff
@@ -0,0 +1,873 @@
+            {
+                throw new ArgumentNullException(nameof(e));
+            }
+
+            return UrlDecodeToBytes(e.GetBytes(str));
+        }
+
+        public static byte[] UrlDecodeToBytes(byte[] bytes, int offset, int count)
+        {
+            if (bytes == null)
+            {
+                return null;
+            }
+
+            if (count == 0)
+            {
+                return new byte[0];
+            }
+
+            int len = bytes.Length;
+            if (offset < 0 || offset >= len)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            }
+
+            if (count < 0 || offset > len - count)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
+
+            MemoryStream result = new MemoryStream();
+            int end = offset + count;
+            for (int i = offset; i < end; i++)
+            {
+                char c = (char)bytes[i];
+                if (c == '+')
+                {
+                    c = ' ';
+                }
+                else if (c == '%' && i < end - 2)
+                {
+                    int xchar = GetChar(bytes, i + 1, 2);
+                    if (xchar != -1)
+                    {
+                        c = (char)xchar;
+                        i += 2;
+                    }
+                }
+
+                result.WriteByte((byte)c);
+            }
+
+            return result.ToArray();
+        }
+
+        public static string UrlEncode(string str)
+        {
+            return UrlEncode(str, Encoding.UTF8);
+        }
+
+        public static string UrlEncode(string str, Encoding e)
+        {
+            if (str == null)
+            {
+                return null;
+            }
+
+            if (str == string.Empty)
+            {
+                return string.Empty;
+            }
+
+            bool needEncode = false;
+            int len = str.Length;
+            for (int i = 0; i < len; i++)
+            {
+                char c = str[i];
+                if ((c < '0') || (c < 'A' && c > '9') || (c > 'Z' && c < 'a') || (c > 'z'))
+                {
+                    if (HttpEncoder.NotEncoded(c))
+                    {
+                        continue;
+                    }
+
+                    needEncode = true;
+                    break;
+                }
+            }
+
+            if (!needEncode)
+            {
+                return str;
+            }
+
+            // avoided GetByteCount call
+            byte[] bytes = new byte[e.GetMaxByteCount(str.Length)];
+            int realLen = e.GetBytes(str, 0, str.Length, bytes, 0);
+            return Encoding.ASCII.GetString(UrlEncodeToBytes(bytes, 0, realLen));
+        }
+
+        public static string UrlEncode(byte[] bytes)
+        {
+            if (bytes == null)
+            {
+                return null;
+            }
+
+            if (bytes.Length == 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)